<a href="https://colab.research.google.com/github/Swarn36/Pandas_excercise/blob/main/Chitpole.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Step-1: Importing necessary libraries

In [ ]:
import numpy as np
import pandas as pd

#Step-2: Importing the dataset

In [ ]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'
chipo = pd.read_csv(url, sep = '\t')

#Step-3: See the first 10 entries

In [ ]:
chipo[:10]

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [ ]:
#or it can be done as follows
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [ ]:
chipo.quantity.sum()

4972

##What is the number of observations?

In [ ]:
chipo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


total `4622` observations

In [ ]:
#name of the columns
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

##What is the number of columns and rows?

In [ ]:
#number of rows in the dataset as well as the observations
chipo.shape[0]

4622

In [ ]:
#number of columns in the dataset
chipo.shape[1]

5

##How is the dataset indexed?

In [ ]:
chipo.index

RangeIndex(start=0, stop=4622, step=1)

##Which one is the most ordered item?

In [ ]:
chipo['item_name'].value_counts().head()

Chicken Bowl           726
Chicken Burrito        553
Chips and Guacamole    479
Steak Burrito          368
Canned Soft Drink      301
Name: item_name, dtype: int64

In [ ]:
c = chipo.groupby(by= 'item_name').sum()
c = c.sort_values(['quantity'], ascending=False)
c.head()

<ipython-input-36-eb405782e5ff>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  c = chipo.groupby(by= 'item_name').sum()


order_id  quantity
item_name                            
Chicken Bowl         713926       761
Chicken Burrito      497303       591
Steak Burrito        328437       386
Canned Soft Drink    304753       351
Steak Bowl           193752       221

`Chicken Bowl` is the most ordered item with `761` numbers

##For the most ordered items, how many items were ordered?

In [ ]:
c['quantity'][0]

761

##What was the most ordered item in the choice_description column?

In [ ]:
chipo.isnull().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

in `choice_description` column there are `1246` NaN values. One should get rid of that values

In [ ]:
chipo.dropna(inplace=True)

In [ ]:
chipo.isnull().sum()

order_id              0
quantity              0
item_name             0
choice_description    0
item_price            0
dtype: int64

In [ ]:
##for the most ordered item in choice_description column
c_cd = chipo.groupby(by = 'choice_description').sum()
c_cd = c_cd.sort_values(['quantity'], ascending=False)
c_cd.head()

<ipython-input-44-bbedb7295c54>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  c_cd = chipo.groupby(by = 'choice_description').sum()


order_id  quantity
choice_description                                                    
[Diet Coke]                                           123455       159
[Coke]                                                122752       143
[Sprite]                                               80426        89
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese...     43088        49
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese...     36041        42

##How many items were orderd in total?

In [ ]:
total_orders = chipo.quantity.sum()
total_orders

3590

##Turn the item price into a float

In [ ]:
chipo.item_price.dtype

dtype('O')

In [ ]:
dollarizer = lambda x: float(x[1:-1])
chipo.item_price = chipo.item_price.apply(dollarizer)
chipo.item_price.dtype

dtype('float64')

##How much was the revenue for the period in the dataset?

In [ ]:
chipo['revenue'] = chipo['item_price'] * chipo['quantity']
chipo.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  revenue  
0                                                NaN        2.39     2.39  
1                                       [Clementine]        3.39     3.39  
2                                            [Apple]        3.39     3.39  
3                                                NaN        2.39     2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98    33.96

In [ ]:
total_revenue = chipo.revenue.sum()
print(f'Total revenue for the period is ${total_revenue}')

Total revenue for the period is $39237.02


##How many orders were made in the period?

In [ ]:
c_oi = chipo.order_id.value_counts()
total_orders = c_oi.count()
print(f'Total orders were {total_orders}')

Total orders were 1834


In [ ]:
chipo.groupby(by = 'order_id').sum()

<ipython-input-74-68d8afb89064>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  chipo.groupby(by = 'order_id').sum()


quantity  item_price  revenue
order_id                               
1                4       11.56    11.56
2                2       16.98    33.96
3                2       12.67    12.67
4                2       21.00    21.00
5                2       13.70    13.70
...            ...         ...      ...
1830             2       23.00    23.00
1831             3       12.90    12.90
1832             2       13.20    13.20
1833             2       23.50    23.50
1834             3       28.75    28.75

[1834 rows x 3 columns]

##What is the average revenue amount per order?

In [ ]:
average_revenue_per_order = total_revenue / total_orders
print(f'Revenue per order ${str(round(average_revenue_per_order,2))}')

Revenue per order $21.39


In [ ]:
#another way to do it
chipo['revenue'] = chipo['quantity'] * chipo['item_price']
order_grouped = chipo.groupby(by=['order_id']).sum()
order_grouped.mean()['revenue']

<ipython-input-71-cb66c8f5a9f9>:3: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  order_grouped = chipo.groupby(by=['order_id']).sum()


21.39423118865867

In [ ]:
order_grouped['revenue'].mean()

21.39423118865867

##How many different items are sold?

In [ ]:
c_item = chipo['item_name'].unique()
c_item

array(['Chips and Fresh Tomato Salsa', 'Izze', 'Nantucket Nectar',
       'Chips and Tomatillo-Green Chili Salsa', 'Chicken Bowl',
       'Side of Chips', 'Steak Burrito', 'Steak Soft Tacos',
       'Chips and Guacamole', 'Chicken Crispy Tacos',
       'Chicken Soft Tacos', 'Chicken Burrito', 'Canned Soda',
       'Barbacoa Burrito', 'Carnitas Burrito', 'Carnitas Bowl',
       'Bottled Water', 'Chips and Tomatillo Green Chili Salsa',
       'Barbacoa Bowl', 'Chips', 'Chicken Salad Bowl', 'Steak Bowl',
       'Barbacoa Soft Tacos', 'Veggie Burrito', 'Veggie Bowl',
       'Steak Crispy Tacos', 'Chips and Tomatillo Red Chili Salsa',
       'Barbacoa Crispy Tacos', 'Veggie Salad Bowl',
       'Chips and Roasted Chili-Corn Salsa',
       'Chips and Roasted Chili Corn Salsa', 'Carnitas Soft Tacos',
       'Chicken Salad', 'Canned Soft Drink', 'Steak Salad Bowl',
       '6 Pack Soft Drink', 'Chips and Tomatillo-Red Chili Salsa', 'Bowl',
       'Burrito', 'Crispy Tacos', 'Carnitas Crispy Tacos

In [ ]:
c_number_diff_item = chipo['item_name'].value_counts().count()
print(f'Total number of different items are {c_number_diff_item}')

Total number of different items are 50


##How many products cost more than $10.00?

In [ ]:
chipo.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  revenue  
0                                                NaN        2.39     2.39  
1                                       [Clementine]        3.39     3.39  
2                                            [Apple]        3.39     3.39  
3                                                NaN        2.39     2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98    33.96

In [ ]:
chipo[chipo['item_price'] > 10.00].value_counts().count()

1116

In [ ]:
chipo[chipo['item_price'] > 10.00].nunique()

order_id              863
quantity                8
item_name              31
choice_description    492
item_price             35
revenue                35
dtype: int64

##What is the price of each item?

In [ ]:
chipo.groupby(by = 'item_name')['item_price'].sum()

item_name
6 Pack Soft Drink                         356.95
Barbacoa Bowl                             672.36
Barbacoa Burrito                          894.75
Barbacoa Crispy Tacos                     120.21
Barbacoa Salad Bowl                       106.40
Barbacoa Soft Tacos                       250.46
Bottled Water                             302.56
Bowl                                       29.60
Burrito                                    44.40
Canned Soda                               137.34
Canned Soft Drink                         438.75
Carnitas Bowl                             736.71
Carnitas Burrito                          597.83
Carnitas Crispy Tacos                      77.96
Carnitas Salad                              8.99
Carnitas Salad Bowl                        66.34
Carnitas Soft Tacos                       375.94
Chicken Bowl                             7342.73
Chicken Burrito                          5575.82
Chicken Crispy Tacos                      472.13
Chicken Sa

In [ ]:
chipo.groupby(by = 'item_name')[['item_price', 'quantity']].value_counts()

item_name          item_price  quantity
6 Pack Soft Drink  6.49        1           53
                   12.98       2            1
Barbacoa Bowl      9.25        1           28
                   11.75       1           20
                   8.99        1            8
                                           ..
Veggie Salad Bowl  8.75        1            8
Veggie Soft Tacos  11.25       1            2
                   8.49        1            2
                   8.75        1            2
                   16.98       2            1
Length: 209, dtype: int64

In [ ]:
chipo[['item_name','item_price']].value_counts()

item_name              item_price
Chips and Guacamole    4.45          349
Chicken Bowl           8.75          313
Canned Soft Drink      1.25          264
Chicken Burrito        8.75          230
Chicken Bowl           11.25         228
                                    ... 
Steak Bowl             17.98           1
                       26.07           1
Carnitas Salad         8.99            1
Carnitas Crispy Tacos  17.98           1
Veggie Soft Tacos      16.98           1
Length: 209, dtype: int64

In [ ]:
chipo['item_name'].value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

##Sort by the name of the item

In [ ]:
chipo.sort_values(by = "item_name")

order_id  quantity          item_name  \
3389      1360         2  6 Pack Soft Drink   
341        148         1  6 Pack Soft Drink   
1849       749         1  6 Pack Soft Drink   
1860       754         1  6 Pack Soft Drink   
2713      1076         1  6 Pack Soft Drink   
...        ...       ...                ...   
2384       948         1  Veggie Soft Tacos   
781        322         1  Veggie Soft Tacos   
2851      1132         1  Veggie Soft Tacos   
1699       688         1  Veggie Soft Tacos   
1395       567         1  Veggie Soft Tacos   

                                     choice_description  item_price  revenue  
3389                                        [Diet Coke]       12.98    25.96  
341                                         [Diet Coke]        6.49     6.49  
1849                                             [Coke]        6.49     6.49  
1860                                        [Diet Coke]        6.49     6.49  
2713                                             [Coke]        6.49     6.49  
...                                                 ...         ...      ...  
2384  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75     8.75  
781   [Fresh Tomato Salsa, [Black Beans, Cheese, Sou...        8.75     8.75  
2851  [Roasted Chili Corn Salsa (Medium), [Black Bea...        8.49     8.49  
1699  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25    11.25  
1395  [Fresh Tomato Salsa (Mild), [Pinto Beans, Rice...        8.49     8.49  

[4622 rows x 6 columns]

##What was the quantity of the most expensive item ordered?

In [ ]:
item_price_quant = chipo.groupby(by = 'item_name')['item_price'].sum()
chipo.sort_values(['revenue'], ascending=False).head()

order_id  quantity                     item_name  \
3598      1443        15  Chips and Fresh Tomato Salsa   
4152      1660        10                 Bottled Water   
1254       511         4               Chicken Burrito   
3602      1443         4               Chicken Burrito   
3887      1559         8                 Side of Chips   

                                     choice_description  item_price  revenue  
3598                                                NaN       44.25   663.75  
4152                                                NaN       15.00   150.00  
1254  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       35.00   140.00  
3602  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       35.00   140.00  
3887                                                NaN       13.52   108.16

In [ ]:
chipo_most = chipo.sort_values(['revenue'], ascending=False).head()
chipo_most[['quantity','item_name']].head(1)

quantity                     item_name
3598        15  Chips and Fresh Tomato Salsa

##How many times was a Veggie Salad Bowl ordered?

In [ ]:
chipo[chipo['item_name'] == 'Veggie Salad'].value_counts().count()

6

##How many times did someone order more than one Canned Soda?

In [ ]:
chipo[(chipo['item_name'] == 'Canned Soda') & (chipo['quantity'] > 1)].value_counts().count()

20

In [ ]:
q = chipo[['item_name','quantity']]
one_canned_soda = q[q['quantity'] > 1]
one_canned_soda[one_canned_soda['item_name'] == 'Canned Soda']

item_name  quantity
18    Canned Soda         2
51    Canned Soda         2
162   Canned Soda         2
171   Canned Soda         2
350   Canned Soda         2
352   Canned Soda         2
698   Canned Soda         2
700   Canned Soda         2
909   Canned Soda         2
1091  Canned Soda         2
1092  Canned Soda         2
1944  Canned Soda         2
2135  Canned Soda         2
2235  Canned Soda         4
2905  Canned Soda         2
2906  Canned Soda         2
3152  Canned Soda         2
3364  Canned Soda         2
3592  Canned Soda         2
3866  Canned Soda         2

In [ ]:
chipo[(chipo['item_name'] == 'Canned Soda') & (chipo['quantity'] > 1)].value_counts()

order_id  quantity  item_name    choice_description  item_price
9         2         Canned Soda  [Sprite]            $2.18         1
23        2         Canned Soda  [Mountain Dew]      $2.18         1
1440      2         Canned Soda  [Diet Coke]         $2.18         1
1349      2         Canned Soda  [Coca Cola]         $2.18         1
1258      2         Canned Soda  [Dr. Pepper]        $2.18         1
1156      2         Canned Soda  [Sprite]            $2.18         1
                                 [Coca Cola]         $2.18         1
901       4         Canned Soda  [Sprite]            $4.36         1
859       2         Canned Soda  [Diet Coke]         $2.18         1
787       2         Canned Soda  [Dr. Pepper]        $2.18         1
450       2         Canned Soda  [Dr. Pepper]        $2.18         1
                                 [Coca Cola]         $2.18         1
376       2         Canned Soda  [Mountain Dew]      $2.18         1
288       2         Canned Soda  [Coca 

##Using Crosstab to visualise easily

In [ ]:
pd.crosstab(chipo['item_name'], chipo['quantity'])

quantity                                1   2   3   4   5   7   8   10  15
item_name                                                                 
6 Pack Soft Drink                       53   1   0   0   0   0   0   0   0
Barbacoa Bowl                           66   0   0   0   0   0   0   0   0
Barbacoa Burrito                        91   0   0   0   0   0   0   0   0
Barbacoa Crispy Tacos                   10   1   0   0   0   0   0   0   0
Barbacoa Salad Bowl                     10   0   0   0   0   0   0   0   0
Barbacoa Soft Tacos                     25   0   0   0   0   0   0   0   0
Bottled Water                          139  12   6   2   1   1   0   1   0
Bowl                                     1   0   1   0   0   0   0   0   0
Burrito                                  6   0   0   0   0   0   0   0   0
Canned Soda                             84  19   0   1   0   0   0   0   0
Canned Soft Drink                      264  27   7   3   0   0   0   0   0
Carnitas Bowl                           66   1   1   0   0   0   0   0   0
Carnitas Burrito                        58   1   0   0   0   0   0   0   0
Carnitas Crispy Tacos                    6   1   0   0   0   0   0   0   0
Carnitas Salad                           1   0   0   0   0   0   0   0   0
Carnitas Salad Bowl                      6   0   0   0   0   0   0   0   0
Carnitas Soft Tacos                     40   0   0   0   0   0   0   0   0
Chicken Bowl                           693  31   2   0   0   0   0   0   0
Chicken Burrito                        521  28   2   2   0   0   0   0   0
Chicken Crispy Tacos                    44   3   0   0   0   0   0   0   0
Chicken Salad                            9   0   0   0   0   0   0   0   0
Chicken Salad Bowl                      97  13   0   0   0   0   0   0   0
Chicken Soft Tacos                     110   5   0   0   0   0   0   0   0
Chips                                  193  17   1   0   0   0   0   0   0
Chips and Fresh Tomato Salsa           105   3   0   1   0   0   0   0   1
Chips and Guacamole                    455  22   1   1   0   0   0   0   0
Chips and Mild Fresh Tomato Salsa        1   0   0   0   0   0   0   0   0
Chips and Roasted Chili Corn Salsa      21   1   0   0   0   0   0   0   0
Chips and Roasted Chili-Corn Salsa      18   0   0   0   0   0   0   0   0
Chips and Tomatillo Green Chili Salsa   41   2   0   0   0   0   0   0   0
Chips and Tomatillo Red Chili Salsa     46   2   0   0   0   0   0   0   0
Chips and Tomatillo-Green Chili Salsa   29   2   0   0   0   0   0   0   0
Chips and Tomatillo-Red Chili Salsa     17   1   2   0   0   0   0   0   0
Crispy Tacos                             2   0   0   0   0   0   0   0   0
Izze                                    20   0   0   0   0   0   0   0   0
Nantucket Nectar                        25   2   0   0   0   0   0   0   0
Salad                                    2   0   0   0   0   0   0   0   0
Side of Chips                           99   0   1   0   0   0   1   0   0
Steak Bowl                             202   8   1   0   0   0   0   0   0
Steak Burrito                          352  14   2   0   0   0   0   0   0
Steak Crispy Tacos                      34   1   0   0   0   0   0   0   0
Steak Salad                              4   0   0   0   0   0   0   0   0
Steak Salad Bowl                        27   2   0   0   0   0   0   0   0
Steak Soft Tacos                        54   1   0   0   0   0   0   0   0
Veggie Bowl                             83   2   0   0   0   0   0   0   0
Veggie Burrito                          94   0   1   0   0   0   0   0   0
Veggie Crispy Tacos                      1   0   0   0   0   0   0   0   0
Veggie Salad                             6   0   0   0   0   0   0   0   0
Veggie Salad Bowl                       18   0   0   0   0   0   0   0   0
Veggie Soft Tacos                        6   1   0   0   0   0   0   0   0

In [ ]:
pd.crosstab(chipo['item_name'], (chipo['quantity'] > 1))

quantity                               False  True 
item_name                                          
6 Pack Soft Drink                         53      1
Barbacoa Bowl                             66      0
Barbacoa Burrito                          91      0
Barbacoa Crispy Tacos                     10      1
Barbacoa Salad Bowl                       10      0
Barbacoa Soft Tacos                       25      0
Bottled Water                            139     23
Bowl                                       1      1
Burrito                                    6      0
Canned Soda                               84     20
Canned Soft Drink                        264     37
Carnitas Bowl                             66      2
Carnitas Burrito                          58      1
Carnitas Crispy Tacos                      6      1
Carnitas Salad                             1      0
Carnitas Salad Bowl                        6      0
Carnitas Soft Tacos                       40      0
Chicken Bowl                             693     33
Chicken Burrito                          521     32
Chicken Crispy Tacos                      44      3
Chicken Salad                              9      0
Chicken Salad Bowl                        97     13
Chicken Soft Tacos                       110      5
Chips                                    193     18
Chips and Fresh Tomato Salsa             105      5
Chips and Guacamole                      455     24
Chips and Mild Fresh Tomato Salsa          1      0
Chips and Roasted Chili Corn Salsa        21      1
Chips and Roasted Chili-Corn Salsa        18      0
Chips and Tomatillo Green Chili Salsa     41      2
Chips and Tomatillo Red Chili Salsa       46      2
Chips and Tomatillo-Green Chili Salsa     29      2
Chips and Tomatillo-Red Chili Salsa       17      3
Crispy Tacos                               2      0
Izze                                      20      0
Nantucket Nectar                          25      2
Salad                                      2      0
Side of Chips                             99      2
Steak Bowl                               202      9
Steak Burrito                            352     16
Steak Crispy Tacos                        34      1
Steak Salad                                4      0
Steak Salad Bowl                          27      2
Steak Soft Tacos                          54      1
Veggie Bowl                               83      2
Veggie Burrito                            94      1
Veggie Crispy Tacos                        1      0
Veggie Salad                               6      0
Veggie Salad Bowl                         18      0
Veggie Soft Tacos                          6      1

In [ ]:
chipo.loc[(chipo['item_name'] == 'Veggie Bowl')][chipo['quantity'] > 1]

<ipython-input-158-3d3819f1f422>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  chipo.loc[(chipo['item_name'] == 'Veggie Bowl')][chipo['quantity'] > 1]


order_id  quantity    item_name  \
846        348         2  Veggie Bowl   
3617      1449         2  Veggie Bowl   

                                     choice_description item_price  
846   [Fresh Tomato Salsa (Mild), [Rice, Sour Cream,...    $16.98   
3617  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...    $16.98

In [ ]:
q_soda = chipo.loc[(chipo['item_name'] == 'Canned Soda')][chipo['quantity'] > 1]
q_soda.quantity.sum()

<ipython-input-160-d11514df08e2>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  q_soda = chipo.loc[(chipo['item_name'] == 'Canned Soda')][chipo['quantity'] > 1]


42